Marlin/Configuration.h

```diff
@@ -433,6 +433,64 @@ const bool Z_MIN_PROBE_ENDSTOP_INVERTING = false; // set to true to invert the l
 //#define Z_PROBE_SLED
 //#define SLED_DOCKING_OFFSET 5 // The extra distance the X axis must travel to pickup the sled. 0 should be fine but you can push it further if you'd like.
 
+// With this option servos are powered only during movement, then turned off to prevent jitter.
+//#define DEACTIVATE_SERVOS_AFTER_MOVE
+
+
+// With probe what is triggered when stowed we can test if it is deployed/stowed or not.
+// This is common for Allen Key probes and some servo probes.
+//#define PROBE_IS_TRIGGERED_WHEN_STOWED_TEST
+
+// Allen key retractable z-probe as seen on many Kossel delta printers - http://reprap.org/wiki/Kossel#Automatic_bed_leveling_probe
+// Deploys by touching z-axis belt. Retracts by pushing the probe down. Uses Z_MIN_PIN.
+//#define Z_PROBE_ALLEN_KEY
+// Allow homing with fake Allen key probe. A debug feature.
+#define TEST_ALLEN_KEY
```

Comment:
The configs do need huge amounts of shifting/reorganizing/renaming.
Im quite sure we agreed not to do that now.
Indeed i don't fell well with introducing the Allen key into all configs now - even if the Cartesian printers would work with them (If you don't try to home to with them).
So for now i'd say - it's our secret.

Despite of that `PROBE_TYPE` is a good idea - especially later on in the code - for example when we check is there is rally only one probe. If it helps in the config? I don't know. All this probe specific settings?

I wonder if we could  have a kit system . Both deploy and stow could consist of a few kind of blocks.
1 test if already deployed/stowed
2 test for homed axis
3 raise
4 remember position
5 triggered when stowed test a
6 pre path
7 move  servo
8 switch pin
9 analog output
10 display message
11 wait for click
12 wait for pin state
13 post path
14 triggered when stowed test b (optional goto 6.)
15 return to stored position
16 switch probe pin check on/off

```
Probetype Blocks
NONE     {}
SERVO    {1, 3, (5), 7, (14), 16}
FIXED    {1, 3, 16}
ALLENKEY {1, 2, 3, 4, 5, 6, 14, 15, 16 }
SLED     {1, 2, 3, 4, (5), 6, 8, 13, (14), 15, 16}
...
```
I guess with this blocks we could handle nearly everything. If not we can add an other type of block.
But it's also not a real simplification for the user, because nearly each of the used blocks has to be configured.